medusa/providers/torrent/html/bjshare.py

```diff
@@ -0,0 +1,250 @@
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Informations about the episode being searched (when not RSS)
+
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        manual_search = "manual_search" in kwargs and kwargs['manual_search']
+        if manual_search:
+            self.max_back_pages = 20
+
+        anime = False
+        if ep_obj and ep_obj.series:
+            anime = ep_obj.series.anime == 1
+
+        search_params = {
+            "order_by": "time",
+            "order_way": "desc",
+            "group_results": 0,
+            "action": "basic",
+            "searchsubmit": 1
+        }
+
+        if 'RSS' in search_strings.keys():
+            search_params["filter_cat[14]"] = 1  # anime
+            search_params["filter_cat[2]"] = 1  # tv shows
+        elif anime:
+            search_params["filter_cat[14]"] = 1
+        else:
+            search_params["filter_cat[2]"] = 1
+
+        for mode in search_strings:
+            items = []
+            log.debug(u"Search Mode: {0}".format(mode))
+
+            # if looking for season, look for more pages
+            if mode == 'Season' and not manual_search:
+                self.max_back_pages = 10
+
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    log.debug(u"Search string: {0}".format(search_string.decode("utf-8")))
+
+                # Remove season / episode from search (not supported by tracker)
+                search_str = re.sub(r'\d+$' if anime else r'[S|E]\d\d', '', search_string).strip()
+                # Remove year from search (not supported by tracker)
+                search_str = re.sub(r"\((\d{4})\)$", '', search_str).strip()
+
+                search_params['searchstr'] = search_str
+
+                next_page = 1
+                has_next_page = True
+                while has_next_page and next_page <= self.max_back_pages:
+
+                    search_params["page"] = next_page
+                    log.debug(u"Page Search: {0}".format(next_page))
+                    next_page += 1
+
+                    response = self.session.get(self.urls['search'], params=search_params)
+                    if not response:
+                        log.debug("No data returned from provider")
+                        continue
+
+                    result = self._parse(response.content, mode, search_string)
+                    has_next_page = result['has_next_page']
+                    items += result['items']
+
+                # For each search mode sort all the items by seeders if available
+                items.sort(key=lambda d: try_int(d.get('seeders', 0)), reverse=True)
```

Comment:
Needs to go as well.

Reply:
Done, it was copied from Alpharatio, should be removed there in future also so...